runtime/onert/backend/train/Tensor.h

```diff
@@ -50,16 +50,21 @@ class TrainableTensor : public Tensor, public train::ITrainableTensor
   TrainableTensor() = delete;
 
 public:
-  TrainableTensor(const ir::OperandInfo &info, const ir::Layout layout) : Tensor{info, layout}
+  TrainableTensor(const ir::OperandInfo &info, const ir::Layout layout)
+    : Tensor{info, layout}, _gradient{info, layout}
   {
     // DO NOTHING
   }
 
 public:
-  void applyGradient(const IPortableTensor &grad_tensor, double lr) override;
+  IPortableTensor &weightTensor() override { return *this; }
```

Comment:
(optional) Not every trainable tensor is weight tensor. Some may be `bias` tensor. Also, do we need this method? TrainableTensor itself is `Tensor`. `grad()` or `gradTensor()` is enough.

Reply:
> Not every trainable tensor is weight tensor. Some may be bias tensor.

Right. It may not be a weight, but may be a bias. I couldn't find a suitable name. Could you recommend?

> Also, do we need this method? TrainableTensor itself is Tensor. grad() or gradTensor() is enough.

Tensors that can be trained need tensors for gradients to be updated. So I put a weight tensor and a gradient tensor together in `TrainableTensor` to make it easier to access them without using index in kernels and optimizers.

Reply:
I don't suggest to use another name. Rather, I suggested not to introduce new method since `TrainableTensor` itself is a `Tensor`, which inherits `Tensor`. Anyway, I don't oppose this PR.